### PR TITLE
Add key ignoring

### PR DESCRIPTION
**Pull request by konradmb**
Added `-k` option to ignore certain key codes.
For example, when you don't want to turn on backlight when pressing arrow keys.
Also, `-d` option shows key values, to be fed to `-k`.

### CODE REVIEW

File: kbd_backlight.cpp [hunk 1]
```diff
@@ -269,21 +273,56 @@ void brightness_control(const std::string &brightnessPath,
...
+void read_events(int devFd, const std::string &brightnessPath,
+				 const std::vector<int> &ignoredKeys, bool showPressedKeys) {
+  int ignoreNextValues = 0;
   while (!end_) {
 	struct input_event ie = {};
 	int rd = read(devFd, &ie, sizeof(struct input_event));
-	if (rd != 0) {
-	  lastEvent_ = std::chrono::system_clock::now();
+	if (rd != 0){
+		if (showPressedKeys && ie.type == 4 && ie.code == 4)
```
Comment (alexmohr) [inline kbd_backlight.cpp]: Please introduce a constant to prevent the usage of this magic number

File: kbd_backlight.cpp [hunk 2]
```diff
@@ -269,21 +273,56 @@ void brightness_control(const std::string &brightnessPath,
...
+		{
+			printf("Pressed key value: %d\n", ie.value);
+			fflush(stdout);
+		}
+		
+		bool correctKey = true;
+		if (ie.type == 4 && ie.code == 4) {
```
Comment (alexmohr) [inline kbd_backlight.cpp]: Use the constant here as well.

File: kbd_backlight.cpp [hunk 3]
```diff
@@ -269,21 +273,56 @@ void brightness_control(const std::string &brightnessPath,
...
+			if (std::find(std::begin(ignoredKeys),
```
Comment (alexmohr) [inline kbd_backlight.cpp]: It would be fast to store the key codes in a std::map to prevent this search.

File: kbd_backlight.cpp [hunk 4]
```diff
@@ -269,21 +273,56 @@ void brightness_control(const std::string &brightnessPath,
...
+			fflush(stdout);
+		}
+		
+		bool correctKey = true;
+		if (ie.type == 4 && ie.code == 4) {
+			if (std::find(std::begin(ignoredKeys), 
+						  std::end(ignoredKeys), 
+						  ie.value) != std::end(ignoredKeys)) {
+				correctKey = false;
+				ignoreNextValues = 2;
```
Comment (alexmohr) [inline kbd_backlight.cpp]: I guess ``ignoreNextValues`` is used to ignore the key up event (1).  Why is it set to 2?

**Review by alexmohr — CHANGES_REQUESTED** [review]
Cool idea but I do have some remarks. 
Please do correct the formatting as well. 
Thanks for your PR

File: kbd_backlight.cpp (hunk 4 above)
Comment (konradmb) [reply kbd_backlight.cpp]: There are 3 events called after every key press, even when holding:
```
Processing key type: 4, code: 4, value: 200
Processing key type: 1, code: 103, value: 1
Processing key type: 0, code: 0, value: 0
```
And when holding key:
```
Processing key type: 4, code: 4, value: 200
Processing key type: 1, code: 103, value: 2
Processing key type: 0, code: 0, value: 0
```

Comment (konradmb) [reply kbd_backlight.cpp]: So, I catch the first scan event, and ignore the two more, that are generated.

Comment (alexmohr) [reply kbd_backlight.cpp]: Good to know. Might be a good idea to add a comment in the code to document this.

**Comment by alexmohr** [comment]
@konradmb Regarding my request to change the vector to a map, is this something you want to solve too?
By the way the formatting still does not fit to the rest of the code, please do not put brackets on a new line after ``if``

**Comment by alexmohr** [comment]
Oh and the readme should include the new option. 😅

**Comment by konradmb** [comment]
>  By the way the formatting still does not fit to the rest of the code, please do not put brackets on a new line after `if`

Oh, yes, didn't see that…

**Comment by konradmb** [comment]
Done!

File: kbd_backlight.cpp [hunk 5]
```diff
@@ -388,6 +435,8 @@ int main(int argc, char **argv) {
 
   std::vector<std::string> inputDevices;
   std::vector<std::string> ignoredDevices;
+  std::map<int, bool> ignoredKeys;
+	bool showPressedKeys = false;
```
Comment (alexmohr) [inline kbd_backlight.cpp]:
```suggestion
  bool showPressedKeys = false;
```

**Review by alexmohr — APPROVED** [review]

**Comment by alexmohr** [comment]
Looks great! Thank you. I'll create a new release in the next couple of days

**Pull request merged**